tools/tinygoize/main.go

```
@@ -2,81 +2,315 @@
+			cg.List = append(cg.List, &ast.Comment{Text: goBuild + constraint})
+
+			if len(f.Comments) > 0 {
+				// insert //go:build after first comment
+				// group, assumed copyright. Doesn't seem
+				// quite right but seems to work.
+				cg.List[0].Slash = f.Comments[0].List[0].Slash + 1
+				f.Comments = append([]*ast.CommentGroup{f.Comments[0], &cg}, f.Comments[1:]...)
+			} else {
+				// prepend //go:build
+				f.Comments = append([]*ast.CommentGroup{&cg}, f.Comments...)
 			}
-			if err := os.WriteFile(file, buf.Bytes(), 0o644); err != nil {
-				log.Fatal(err)
+		}
+
+		// Complete source file.
+		var buf bytes.Buffer
+		if err = p.Fprint(&buf, fset, f); err != nil {
+			log.Fatalf("Printing:%v", err)
+		}
+		if err := os.WriteFile(file, buf.Bytes(), 0o644); err != nil {
+			log.Fatal(err)
+		}
+	}
+	return
+}
+
+func writeMarkdown(file *os.File, pathMD *string, tinygo *string, status BuildStatus) (err error) {
+
+	fmt.Fprintf(file, "---\n\n")
+	fmt.Fprintf(file, "DO NOT EDIT.\n\n")
+	fmt.Fprintf(file, "Generated via `go run tools/tinygoize/main.go`\n\n")
+	fmt.Fprintf(file, "%v\n\n", tinygoVersion(tinygo))
+	fmt.Fprintf(file, "---\n\n")
+
+	fmt.Fprintf(file, `# Status of u-root + tinygo
+This document aims to track the process of enabling all u-root commands
+to be built using tinygo. It will be updated as more commands can be built via:
+
+    u-root> go run tools/tinygoize/main.go cmds/{core,exp,extra}/*
+
+Commands that cannot be built with tinygo have a \"(!tinygo || tinygo.enable)\"
+build constraint. Specify the "tinygo.enable" build tag to attempt to build
+them.
+
+    tinygo build -tags tinygo.enable cmds/core/ls
+
+The list below is the result of building each command for Linux, x86_64.
+
+The necessary additions to tinygo will be tracked in
+[#2979](https://github.com/u-root/u-root/issues/2979).
+
+---
+
+## Commands Build Status
+`)
+
+	linkText := func(dir string) string {
+		relPath, _ := filepath.Rel(filepath.Dir(*pathMD), dir)
```

Why ignore the error here?

---

Added comment.